tests/Spectra.py

```diff
@@ -371,6 +369,49 @@ def testGetSpectra(self):
             spectra = specSet.getSpectra()
             self.assertEqual(spectra[0].getSpectrum().shape[0], length)
 
+    def testProfile(self):
+        """
+        Test that there are no systematics in the profile
+        We compare the original FiberTrace images to the reconstructed ones.
+        The reconstructed images are calculated from the spatial profile images
+        and the extracted spectra.
+        We sort the ratio of both images (original / recreated) from all
+        FiberTraces by the distance of the pixel to the FiberTrace's center,
+        bin them in small bins (10 times the oversampling rate used for the
+        profile calculation), and calculate the mean and standard deviation
+        for each bin. The test passes if 1.0 is within 1 standard deviation
+        of the mean value.
+        """
+        nBinsPerOverSampleStep = 10.
+        myFindTask = fataTask.FindAndTraceAperturesTask()
+        myFindTask.config.xLow = -5.
+        myFindTask.config.xHigh = 5.
+        fts = myFindTask.run(self.flat)
+
+        myProfileTask = cfftpTask.CreateFlatFiberTraceProfileTask()
+        myProfileTask.config.overSample = 10
+
+        for interpolation in ['SPLINE3', 'PISKUNOV']:
+            myProfileTask.config.profileInterpolation = interpolation
+            myProfileTask.run(fts)
+
+            distanceFromCenterOrigProfRec = drpStellaMath.getDistTraceProfRec(fts)
+            dist, orig, rec = [distanceFromCenterOrigProfRec[:,0],
+                               distanceFromCenterOrigProfRec[:,1],
+                               distanceFromCenterOrigProfRec[:,3]]
+            ratio = orig / np.where(rec < 0.1, 1., rec)
```

Comment:
Added comment why we do this. Essentially we avoid division by zero and multiplying the original pixel values with a large number. As there is no signal in these pixels, the SNR is gonna be small here anyway, so all those pixels are useless for flat-fielding